pkg/api/types.go

```diff
@@ -150,6 +150,11 @@ type ObjectMeta struct {
 	// from the list. If the deletionTimestamp of the object is non-nil, entries
 	// in this list can only be removed.
 	Finalizers []string `json:"finalizers,omitempty"`
+
+	// The cluster which the object belongs to. The federation apiserver/controllers use this to
+	// distinguish the object at the federation level from it's constituents at the cluster level.
+	// An empty value means the object belongs to the local cluster (the one the client connects to).
```

Comment:
Is the federation-controller always supposed to be in the form of 2 pods, federation-apiserver pod and federation-controller-manager pod, in one of the federated clusters? If it is deployed on a separate VM, e.g., controller VM, it means the controller does not belong to any cluster. So, no local cluster exists and an RS with an empty cluster id value cannot be created.